packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -3,50 +3,343 @@
+///   void initState() {
+///     super.initState();
+///     _node = FocusNode(debugLabel: 'Button');
+///     _nodeAttachment = _node.attach(context, onKey: _handleKeyPress);
+///   }
+///
+///   bool _handleKeyPress(FocusNode node, RawKeyEvent event) {
+///     if (event is RawKeyDownEvent) {
+///       print('Focus node ${node.debugLabel} got key event: ${event.logicalKey}');
+///       if (event.logicalKey == LogicalKeyboardKey.keyR) {
+///         print('Changing color to red.');
+///         setState(() {
+///           _color = Colors.red;
+///         });
+///         return true;
+///       } else if (event.logicalKey == LogicalKeyboardKey.keyG) {
+///         print('Changing color to green.');
+///         setState(() {
+///           _color = Colors.green;
+///         });
+///         return true;
+///       } else if (event.logicalKey == LogicalKeyboardKey.keyB) {
+///         print('Changing color to blue.');
+///         setState(() {
+///           _color = Colors.blue;
+///         });
+///         return true;
+///       }
+///     }
+///     return false;
+///   }
+///
+///   @override
+///   void dispose() {
+///     // The attachment will automatically be detached in dispose().
+///     _node.dispose();
+///     super.dispose();
+///   }
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     _nodeAttachment.reparent(Focus.of(context));
+///     return RaisedButton(
+///       color: _node.hasFocus ? _color : null,
+///       onPressed: () {
+///         if (_node.hasFocus) {
+///           setState(() {
+///             _node.unfocus();
+///           });
+///         } else {
+///           setState(() {
+///             _node.requestFocus();
+///           });
+///         }
+///       },
+///       child: Text(_node.hasFocus ? "I'm in color! Press R,G,B!" : 'Press to focus'),
+///     );
+///   }
+/// }
+/// ```
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   final TextTheme textTheme = Theme.of(context).textTheme;
+///   return DefaultTextStyle(
+///     style: textTheme.display1,
+///     child: ColorfulButton(),
+///   );
+/// }
+/// ```
+/// {@end-tool}
 ///
 /// See also:
 ///
-///  * [FocusScopeNode], which is an interior node in the focus tree.
-///  * [FocusScope.of], which provides the [FocusScopeNode] for a given
-///    [BuildContext].
-class FocusNode extends ChangeNotifier {
-  FocusScopeNode _parent;
+///   * [Focus], a widget that manages a [FocusNode] and provides access to
+///     focus information and actions to its descendant widgets.
+///   * [FocusScope], a widget that manages a [FocusScopeNode] and provides
+///     access to scope information and actions to its descendant widgets.
+///   * [FocusAttachment], a widget that connects a [FocusScopeNode] to the
+///     focus tree.
+///   * [FocusManager], a singleton that manages the focus and distributes key
+///     events to focused nodes.
+class FocusNode with DiagnosticableTreeMixin, ChangeNotifier {
```

Comment:
i feel like these docs are still missing a discussion of the various actors and what parts of this API surface are relevant to each one. Something like:

`There are several actors involved in the lifecycle of a [FocusNode]. They are created and disposed by their _owner_, attached, detached, and reparented by their _host_ (which must be the [State] of a [StatefulWidget]), and managed by the [FocusManager]. Different parts of the [FocusNode] API are intended for these different actors.`

Then for each API doc below, specify whether it's intended for the owner or the host (or, in some cases, either, though hopefully that's only for read-only APIs).